airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -450,6 +450,60 @@ def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapp
                 self.logger.warn(f"Unexpected error in `parse_ersponse`: {e}, the actual response data: {response.text}")
                 yield {}
 
+           
+class OrdersGraphQl(IncrementalShopifyStream):
+    filter_field = "createdAt"
+    cursor_field = "createdAt"
+    data_field = "graphql"
+    http_method = "POST"
+    limit = 20
+
+    def path(self, **kwargs) -> str:
+        return f"{self.data_field}.json"
+
+    def request_params(
+        self,
+        stream_state: Optional[Mapping[str, Any]] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+        **kwargs,
+    ) -> MutableMapping[str, Any]:
+        return {}
+
+    def request_body_json(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Optional[Mapping[str, Any]] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Optional[Mapping]:
+        state_value = stream_state.get(self.filter_field)
+        if state_value:
+            filter_value = state_value
+        else:
+            filter_value = self.default_filter_field_value
+        query = get_query_orders(
+            first=self.limit, filter_field=self.filter_field, filter_value=filter_value, next_page_token=next_page_token
+        )
+        return {"query": query}
+
+    @staticmethod
+    def next_page_token(response: requests.Response) -> Optional[Mapping[str, Any]]:
+        page_info = response.json()["data"]["orders"]["pageInfo"]
+        has_next_page = page_info["hasNextPage"]
+        if has_next_page:
+            return page_info["endCursor"]
+        else:
+            return None
+
+    @limiter.balance_rate_limit(api_type=ApiTypeEnum.graphql.value)
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        if response.status_code is requests.codes.OK:
+            try:
+                json_response = response.json()["data"]["orders"]["nodes"]
+                yield from self.produce_records(json_response)
+            except RequestException as e:
+                self.logger.warn(f"Unexpected error in `parse_ersponse`: {e}, the actual response data: {response.text}")
```

Comment:
I spent lots of time in previous Shopify to find an exception in logic, when was an empty response when errror appears in the place. Maybe throw an exception? We will see it in Centry and will fix.

Reply:
This service has balancer and checker